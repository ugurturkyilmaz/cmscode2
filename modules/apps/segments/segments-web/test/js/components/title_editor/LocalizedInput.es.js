/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, fireEvent, render} from '@testing-library/react';
import React from 'react';

import LocalizedInput from '../../../../src/main/resources/META-INF/resources/js/components/title_editor/LocalizedInput.es';

const LOCALIZED_DROPDOWN_BUTTON = 'localized-dropdown-button';

const LOCALIZED_MAIN_INPUT_ID = 'localized-main-input';

const PRE_EXISTING_VALUE = 'Testing default value';

const PRE_EXISTING_VALUE_ALT = 'Testing alternative default value';

describe('LocalizedInput', () => {
	describe('with minimal configuration', () => {
		afterEach(cleanup);

		it('renders', () => {
			const {asFragment} = setUpComponent();
			expect(asFragment()).toMatchSnapshot();
		});

		it('has main input with empty value', () => {
			const {getByTestId} = setUpComponent();
			const mainInput = getByTestId(LOCALIZED_MAIN_INPUT_ID);

			expect(mainInput.value).toBeFalsy();
		});

		function setUpComponent() {
			const {asFragment, getByTestId} = render(
				<LocalizedInput
					availableLanguages={{
						en_US: '',
						es_ES: '',
					}}
					initialLanguageId="en_US"
				/>
			);

			return {asFragment, getByTestId};
		}
	});

	describe('rendering with pre-existing value', () => {
		afterEach(cleanup);

		it('renders', () => {
			const {asFragment} = setUpComponent();
			expect(asFragment()).toMatchSnapshot();
		});

		it('has main input with value', () => {
			const {getByTestId} = setUpComponent();
			const mainInput = getByTestId(LOCALIZED_MAIN_INPUT_ID);
			expect(mainInput.value).toBe(PRE_EXISTING_VALUE);
		});

		it('has dropdown with options', () => {
			const {asFragment, getByTestId} = setUpComponent();
			const dropdownButton = getByTestId(LOCALIZED_DROPDOWN_BUTTON);
			fireEvent.click(dropdownButton);

			expect(asFragment()).toMatchSnapshot();
		});

		it('switches main input value on language selection', () => {
			const {getByTestId, getByText} = setUpComponent();
			const dropdownButton = getByTestId(LOCALIZED_DROPDOWN_BUTTON);
			const mainInput = getByTestId(LOCALIZED_MAIN_INPUT_ID);

			expect(mainInput.value).toBe(PRE_EXISTING_VALUE);

			fireEvent.click(dropdownButton);

			const spanishSelector = getByText('es-ES');
			fireEvent.click(spanishSelector);

			expect(mainInput.value).toBe(PRE_EXISTING_VALUE_ALT);
		});

		function setUpComponent() {
			const testHelpers = render(
				<LocalizedInput
					availableLanguages={{
						en_US: '',
						es_ES: '',
					}}
					initialLanguageId="en_US"
					initialValues={{
						en_US: PRE_EXISTING_VALUE,
						es_ES: PRE_EXISTING_VALUE_ALT,
					}}
				/>
			);

			return testHelpers;
		}
	});
});
