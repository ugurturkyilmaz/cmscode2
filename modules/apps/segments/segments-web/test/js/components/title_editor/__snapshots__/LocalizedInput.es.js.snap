// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalizedInput rendering with pre-existing value has dropdown with options 1`] = `
<DocumentFragment>
  <div
    class="input-group input-localized input-localized-input"
  >
    <div
      class="dropdown postion-relative lfr-icon-menu open"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="btn btn-monospaced btn-secondary dropdown-toggle"
        data-testid="localized-dropdown-button"
        role="button"
        title=""
        type="button"
      >
        <span
          class="inline-item"
        >
          <svg
            class="lexicon-icon lexicon-icon-en-us"
            role="presentation"
          >
            <use
              xlink:href="#en-us"
            />
          </svg>
        </span>
        <span
          class="btn-section"
        >
          en-US
        </span>
      </button>
      <ul
        class="d-block dropdown-menu"
        role="menu"
      >
        <li
          role="presentation"
        >
          <span
            class="dropdown-item lfr-icon-item palette-item taglib-icon"
            role="menuitem"
            tabindex="0"
            target="_self"
          >
            <span
              class="inline-item inline-item-before"
            >
              <svg
                class="lexicon-icon lexicon-icon-en-us"
                role="presentation"
              >
                <use
                  xlink:href="#en-us"
                />
              </svg>
            </span>
            <span
              class="taglib-text-icon"
            >
              en-US
              <span
                class="label label-success ml-1"
              >
                translated
              </span>
            </span>
          </span>
        </li>
        <li
          role="presentation"
        >
          <span
            class="dropdown-item lfr-icon-item palette-item taglib-icon"
            role="menuitem"
            tabindex="0"
            target="_self"
          >
            <span
              class="inline-item inline-item-before"
            >
              <svg
                class="lexicon-icon lexicon-icon-es-es"
                role="presentation"
              >
                <use
                  xlink:href="#es-es"
                />
              </svg>
            </span>
            <span
              class="taglib-text-icon"
            >
              es-ES
              <span
                class="label label-success ml-1"
              >
                translated
              </span>
            </span>
          </span>
        </li>
      </ul>
    </div>
    <div
      class="input-group-item ml-2"
    >
      <input
        class="field form-control form-control-inline language-value rounded"
        data-testid="localized-main-input"
        placeholder=""
        type="text"
        value="Testing default value"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`LocalizedInput rendering with pre-existing value renders 1`] = `
<DocumentFragment>
  <div
    class="input-group input-localized input-localized-input"
  >
    <div
      class="dropdown postion-relative lfr-icon-menu "
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="btn btn-monospaced btn-secondary dropdown-toggle"
        data-testid="localized-dropdown-button"
        role="button"
        title=""
        type="button"
      >
        <span
          class="inline-item"
        >
          <svg
            class="lexicon-icon lexicon-icon-en-us"
            role="presentation"
          >
            <use
              xlink:href="#en-us"
            />
          </svg>
        </span>
        <span
          class="btn-section"
        >
          en-US
        </span>
      </button>
    </div>
    <div
      class="input-group-item ml-2"
    >
      <input
        class="field form-control form-control-inline language-value rounded"
        data-testid="localized-main-input"
        placeholder=""
        type="text"
        value="Testing default value"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`LocalizedInput with minimal configuration renders 1`] = `
<DocumentFragment>
  <div
    class="input-group input-localized input-localized-input"
  >
    <div
      class="dropdown postion-relative lfr-icon-menu "
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="btn btn-monospaced btn-secondary dropdown-toggle"
        data-testid="localized-dropdown-button"
        role="button"
        title=""
        type="button"
      >
        <span
          class="inline-item"
        >
          <svg
            class="lexicon-icon lexicon-icon-en-us"
            role="presentation"
          >
            <use
              xlink:href="#en-us"
            />
          </svg>
        </span>
        <span
          class="btn-section"
        >
          en-US
        </span>
      </button>
    </div>
    <div
      class="input-group-item ml-2"
    >
      <input
        class="field form-control form-control-inline language-value rounded"
        data-testid="localized-main-input"
        placeholder=""
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
